tests/farmer_harvester/test_farmer.py

```diff
@@ -0,0 +1,519 @@
+            target_puzzle_hash=bytes32.fromhex("344587cf06a39db471d2cc027504e8688a0a67cce961253500c956c73603fd58"),
+            p2_singleton_puzzle_hash=p2_singleton_puzzle_hash,
+            owner_public_key=G1Element.from_bytes(
+                bytes.fromhex(
+                    "8348455278ecec68325b6754b1f3218cde1511ca9393197e7876d7ae04af1c4dd86b0c50601cf5daeb034e8f7c226537"
+                )
+            ),
+        )
+        expected_pool_state = expected_pool_stats.copy()
+        expected_pool_state["p2_singleton_puzzle_hash"] = p2_singleton_puzzle_hash.hex()
+        expected_pool_state["pool_config"] = pool_config
+
+        return NewProofOfSpaceCase(
+            difficulty=difficulty,
+            sub_slot_iters=sub_slot_iters,
+            challenge_hash=bytes32.fromhex("9dcc155cafa8723627d01d00118767269a15f5f0d496dc9c2940a45b63c0c00c"),
+            sp_hash=bytes32.fromhex("bca3ac17a0a8fc072188b3738455a3f006528d181c39e31c88e8755492575272"),
+            plot_identifier="test",
+            signage_point_index=uint8(0),
+            plot_id=bytes32.fromhex("baaa6780c53d4b3739b8807b4ae79a76644ddf0d9e03dc7d0a6a0e613e764d9f"),
+            plot_size=uint8(32),
+            plot_challenge=bytes32.fromhex("7580e4c366dc2c94c37ce44943f9629a3cd6e027d7b24cd014adeaa578d4b0a2"),
+            plot_public_key=G1Element.from_bytes(
+                bytes.fromhex(
+                    "a6126295fbf0f50dbed8dc41e236241413fdc8a97e650e3ed69d66d0921d3236f8961cc1cf8c1b195521c2d9143048e2"
+                )
+            ),
+            pool_public_key=None,
+            pool_contract_puzzle_hash=p2_singleton_puzzle_hash,
+            height=uint32(1),
+            proof=bytes.fromhex(
+                "2aa93f8c112c274d3c707d94d0a0e07b6710d96fca284baa198ca4632ee1f591edad47d97e854aec76dc08519614a0d17253fc24fbaab66f558b2f6afd2210acac09b48e01fd0b8334eae94d55db5df28c80c03a586ce8afe3b986e6aceef4d493a7c0b1bf30baaafedc1ef9af6a1eee911e3873e9229a0bf18cfc95bcfc3dac36522c84a85fd16a3ff501b44024e371bbf7a3ee9d4ded6e05c82e95083ea5ead2d64e4a46e0550e98fc56b30760b4bcf439ac0ee675e157fcfecd4b294c6d14987883e3659777c68d6a6962c770a66817f4dc641b5823b077093df10da031c14b32005bf38eb53944bf3bc5a1d19c21e3b96759b7f557b4668687b7a8a36344"
+            ),
+            pool_config=pool_config,
+            pool_difficulty=pool_difficulty,
+            authentication_token_timeout=authentication_token_timeout,
+            use_invalid_peer_response=use_invalid_peer_response,
+            expected_pool_state=expected_pool_state,
+        )
+
+
+@pytest.mark.parametrize(
+    argnames="case",
+    argvalues=[
+        pytest.param(StripOldEntriesCase([], 0, []), id="no_params"),
+        pytest.param(StripOldEntriesCase([(1689491043.3493967, 1)], 1689491044, []), id="stripped"),
+        pytest.param(
+            StripOldEntriesCase([(1689491043.3493967, 1)], 1689491043, [(1689491043.3493967, 1)]),
```

Comment:
I would recommend using smaller values to demonstrate this behavior. say, single digit timestamps. it would be a lot easier to read and be confident that the behavior is correct.

Reply:
Resolving comments in test code - we should iterate and simplfy post release